### ISSUE TITLE
Tree Control Node does not update to show deleted TreeItems immediately

### ISSUE THREAD

**Issue by ShatReal** [godotengine/godot#69553]
### Godot version

3.5.1

### System information

Windows 11

### Issue description

After freeing TreeItem it still is there in Tree.

### Steps to reproduce

```
extends Tree


func _ready() -> void:
	var root := create_item()
	var child := create_item(root)
	child.set_text(0, "delete")
	yield(get_tree().create_timer(0.1), "timeout")
	child.free()
```

### Minimal reproduction project

N/A

**Comment by ShatReal** [comment]
Well, calling TreeItem.remove_child() updates it.

**Comment by Hiiamwilliam** [comment]
If you call `update` (which is from `CanvasItem`) it should redraw correctly after freeing.

**Comment by akien-mga** [comment]
Fixed by #69569.
